### PR TITLE
Update JS SDK section in docs

### PR DESCRIPTION
**Pull request by AnastasiiaVashchuk**

### CODE REVIEW

File: docs/api/js/types.md [hunk 1]
```diff
@@ -48,6 +49,7 @@ export type BlockTag =
     | 'earliest'
     | 'pending';
 
+// TODO (SMA-1585): Support create2 variants.
```
Comment (Deniallugo) [inline docs/api/js/types.md:52]: Seems, that we shouldn't mention TODOs in docs

File: docs/api/js/types.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ All the types which are used in the SDK are referenced here:
 
 ```typescript
 import { BytesLike, BigNumberish, providers, BigNumber } from 'ethers';
```
Comment (Deniallugo) [inline docs/api/js/types.md:6]: Why not just provide a link to our sdk, once it will be public.

File: docs/api/js/utils.md [hunk 3]
```diff
@@ -38,17 +43,35 @@ export const IERC20 = new utils.Interface(require("../abi/IERC20.json"));
 Utility method that returns the correctly formed `paymasterParams` object for the common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
 
 ```typescript
-export function getPaymasterParams(paymasterAddress: Address, paymasterInput: PaymasterInput): PaymasterParams;
+export function getPaymasterParams(paymasterAddress: Address, paymasterInput: PaymasterInput): PaymasterParams {
```
Comment (Deniallugo) [inline docs/api/js/utils.md:46]: I'd say that implementation details are redundant here. the definition is enough

File: docs/api/js/accounts-l1-l2.md [hunk 4]
```diff
@@ -191,25 +192,18 @@ async requestExecute(transaction: {
...
-| transaction.refundRecipient(optional)         |  
-|--                                                                                                                                                         
-| returns                               | `PriorityOpResponse` object.                                                                                                                                                                                                                                                                                                                                     
+| Name                                     | Description                                                                                                                                     |
+| ---------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------- |
+| transaction.contractAddress              | The L2 receiver address.                                                                                                                         |
```
Comment (StanislavBreadless) [inline docs/api/js/accounts-l1-l2.md:197]: it is not "receiver", it is a contract to be called

File: docs/api/js/accounts-l1-l2.md [hunk 5]
```diff
@@ -191,25 +192,18 @@ async requestExecute(transaction: {
...
+| transaction.calldata                     | The input of the L2 transaction.                                                                                                                 |
+| transaction.l2GasLimit                   | Maximum amount of L2 gas that transaction can consume during execution on L2.                                                                    |
+| transaction.l2Value?           | `msg.value` of L2 transaction.                                                                                                                   |
+| transaction.factoryDeps?       | An array of L2 bytecodes that will be marked as known on L2.                                                                                     |
+| transaction.operatorTip?       | If the ETH value passed with the transaction is not explicitly stated in the overrides, this field will be equal to the tip the operator will receive on top of the base cost of the transaction. |
```
Comment (StanislavBreadless) [inline docs/api/js/accounts-l1-l2.md:202]: Please introduce a note (similar to the one that was here before) that this field has not meaning as of now

File: docs/api/js/accounts-l1-l2.md [hunk 6]
```diff
@@ -191,25 +192,18 @@ async requestExecute(transaction: {
...
+| transaction.l2GasLimit                   | Maximum amount of L2 gas that transaction can consume during execution on L2.                                                                    |
+| transaction.l2Value?           | `msg.value` of L2 transaction.                                                                                                                   |
+| transaction.factoryDeps?       | An array of L2 bytecodes that will be marked as known on L2.                                                                                     |
+| transaction.operatorTip?       | If the ETH value passed with the transaction is not explicitly stated in the overrides, this field will be equal to the tip the operator will receive on top of the base cost of the transaction. |                                                                                                                                         
+| transaction.gasPerPubdataByte?  | The maximum amount L2 gas that the operator may charge the user for.                                                                            |
```
Comment (StanislavBreadless) [inline docs/api/js/accounts-l1-l2.md:203]: for what?

File: docs/api/js/accounts.md [hunk 7]
```diff
@@ -111,29 +110,23 @@ const wallet = unconnectedWallet.connect(provider).connectToL1(ethProvider);
...
+    erc20: IL1Bridge;
+}>
 ```
 
 #### Inputs and outputs
 
 | Name    | Description                                      |
 | ------- | ------------------------------------------------ |
-| returns | `Contract` wrapper of the zkSync smart contract. |
+| returns | zkSync `IL1Bridge` contract `ethers` instance. |
```
Comment (StanislavBreadless) [inline docs/api/js/accounts.md:122]: why did you remove the description for a method to get the main contract?

File: docs/api/js/accounts.md [hunk 8]
```diff
@@ -410,13 +408,13 @@ async transfer(tx: {
...
-| tx.amount            | The amount of the token to transfer.                                |
-| token (optional)     | The address of the token. `ETH` by default.                         |
-| overrides (optional) | Transaction overrides, such as `nonce`, `gasLimit` etc. |
-| returns              | An `ethers.ContractTransaction` object.                             |
+| Name                 | Description                                             |
+| -------------------- | ------------------------------------------------------- |
+| tx.to                | The address of the recipient.                           |
+| tx.amount            | The amount of the token to transfer.                    |
+| token?     | The address of the token. `ETH` by default.             |
+| overrides? | **zkSync** transaction overrides. May be used to pass l2 `gasLimit`, `gasPrice`, etc.    |
```
Comment (StanislavBreadless) [inline docs/api/js/accounts.md:416]:
```suggestion
| overrides? | **zkSync** transaction overrides. May be used to pass L2 `gasLimit`, `gasPrice`, etc.    |
```

File: docs/api/js/features.md [hunk 9]
```diff
@@ -15,34 +15,37 @@ This document will focus solely on how to pass these arguments to the SDK.
...
+`ethers.js` has a notion of overrides. For any on-chain transaction, `ethers.js` finds the optimal `gasPrice`, `gasLimit`, `nonce`, and other important fields under the hood. But sometimes, you may have a need to explicitly provide these values (you want to set a smaller `gasPrice` for instance, or sign a transaction with future `nonce`).
 
 In this case, you can provide an `Overrides` object as the last parameter. There you can supply fields like `gasPrice`, `gasLimit`, `nonce` etc.
 
-In order to make the SDK as flexible as possible, the library uses the overrides to supply zkSync-specific fields. To supply zkSync-specific fields, you need to pass the following override:
+In order to make the SDK as flexible as possible, `zksync-web3` uses `customData` object in the overrides to supply zkSync-specific fields.To supply zkSync-specific fields, you need to pass the following override:
```
Comment (StanislavBreadless) [inline docs/api/js/features.md:22]:
```suggestion
In order to make the SDK as flexible as possible, `zksync-web3` uses `customData` object in the overrides to supply zkSync-specific fields. To supply zkSync-specific fields, you need to pass the following override:
```

File: docs/api/js/getting-started.md [hunk 10]
```diff
@@ -122,10 +123,10 @@ To track the status of this transaction:
 
 ```typescript
 // Await commitment
-const transferReceipt = await transfer.wait();
+const committedTxReceipt = await transfer.wait();
```
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply docs/api/js/getting-started.md:126]: @Deniallugo not sure if it is litterally same, but maybe it is worth it to not repeat the same local variable name in one snippet

File: docs/api/js/getting-started.md [hunk 11]
```diff
@@ -58,21 +58,21 @@ To control your account in zkSync, use the `zksync.Wallet` object. It can sign t
...
 ```typescript
-const deposit = await syncWallet.deposit({
+const deposit = await zkSyncWallet.deposit({
   token: zksync.utils.ETH_ADDRESS,
   amount: ethers.utils.parseEther("1.0"),
 });
 ```
 
-**NOTE:** Each token inside zkSync has an address. If ERC-20 tokens are being bridged, you should supply the token's L1 address in the `deposit` function, or zero address (`0x0000000000000000000000000000000000000000`) if you want to deposit ETH. Note, that for the ERC-20 tokens the address of their corresponding L2 token will be different from the one on Ethereum.
+**NOTE:** Each token inside zkSync has an address. If `ERC-20`      tokens are being bridged, you should supply the token's L1 address in the `deposit` function, or zero address (`0x0000000000000000000000000000000000000000`) if you want to deposit ETH. Note, that for the `ERC-20`      tokens the address of their corresponding L2 token will be different from the one on Ethereum.
```
Comment (StanislavBreadless) [inline docs/api/js/getting-started.md:75]:
```suggestion
**NOTE:** Each token inside zkSync has an address. If `ERC-20` tokens are being bridged, you should supply the token's L1 address in the `deposit` function, or zero address (`0x0000000000000000000000000000000000000000`) if you want to deposit ETH. Note, that for the `ERC-20`      tokens the address of their corresponding L2 token will be different from the one on Ethereum.
```

File: docs/api/js/types.md (hunk 2 above)
Comment (StanislavBreadless) [reply docs/api/js/types.md:6]: @Deniallugo what do you mean? These objects can only be imported from `ethers`. `ethers` is a peer dependency for our SDK anyway

File: docs/api/js/utils.md [hunk 12]
```diff
@@ -13,7 +13,12 @@ Most of the utilities are used internally by the zkSync team. So this document w
 
 ## The "address" of ether
 
-While formally ether is a token with address `0x000000000000000000000000000000000000800a` on zkSync, we use the "zero address" as a more user-friendly alias:
+While *de facto* ether is a token with address
+
+```typescript
+export const L2_ETH_TOKEN_ADDRESS = "0x000000000000000000000000000000000000800a";
+```
+ on zkSync, *de jure* in SDK and API "zero address" is used as a more user-friendly alias:
```
Comment (StanislavBreadless) [inline docs/api/js/utils.md:21]: we are not a country, there is no *jure* here : )

File: docs/api/js/utils.md [hunk 13]
```diff
@@ -38,17 +43,35 @@ export const IERC20 = new utils.Interface(require("../abi/IERC20.json"));
...
+## Useful gas constants
 Currently, there is no method to accurately estimate the required `gasPerPubdataLimit`. That's why for now, it is highly recommended to provide the `DEFAULT_GAS_PER_PUBDATA_LIMIT`. Users are not charged more by providing it.
 Later on it will be possible to query the current recommended limit.
 
 ```typescript
-const GAS_PER_PUBDATA_BYTE = 16;
-export const DEFAULT_GAS_PER_PUBDATA_LIMIT = GAS_PER_PUBDATA_BYTE * 10_000;
+const GAS_PER_PUBDATA_BYTE = 17;
+
+export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 800;
```
Comment (StanislavBreadless) [inline docs/api/js/utils.md:70]: Please update according to the latest version of the SDK (a new one was released on Friday)

File: docs/api/js/types.md (hunk 2 above)
Comment (AnastasiiaVashchuk) [reply docs/api/js/types.md:6]: If I understood you correctly, even once it will be public,  SDK should be documented, in ideally every function, type, etc. Now it is not public.

File: docs/dev/tutorials/custom-aa-tutorial.md [hunk 14]
```diff
@@ -6,6 +6,11 @@ In this tutorial, we build a factory that deploys 2-of-2 multisig accounts.
 <TocHeader />
 <TOC class="table-of-contents" :include-level="[2,3]" />
 
+::: warning
+
+Please note, that in the new `0.13.0` SDK version, the API/Node layer operates with gas. The ergs concept is used by VM only.
```
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply docs/dev/tutorials/custom-aa-tutorial.md:11]: I guess it is fine for the transaction period so that those familiar with the previous version of zkSync knew what is going on. 

@AnastasiiaVashchuk please make sure to use `0.13.1` version as `0.13.0` is outdated

File: docs/api/js/utils.md (hunk 12 above)
Comment (AnastasiiaVashchuk) [reply docs/api/js/utils.md:21]: should I say ` formally`?

File: docs/api/js/accounts-l1-l2.md [hunk 15]
```diff
@@ -28,12 +28,12 @@ async approveERC20(
...
+| Name                 | Description                                                                                                                                                                                                      |
+| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| token                | The Ethereum address of the token.                                                                                                                                                                               |
+| amount               | The amount of the token to be approved.                                                                                                                                                                          |
+| overrides? | **Ethereum** transaction overrides. May be used to pass `gasLimit`, `gasPrice`, etc. You can also provide a custom address of the L1 bridge to use (the bridge provided by the `Matter Labs` team is used by default). |
```
[the remark this answers is not in the document]
Comment (AnastasiiaVashchuk) [reply docs/api/js/accounts-l1-l2.md:35]: @StanislavBreadless  what do you think?
since it is **js** SDK it is clean from the context what is `?`
 writing `(optional)` everywhere, looks ugly

File: docs/api/js/getting-started.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (AnastasiiaVashchuk) [reply docs/api/js/getting-started.md:126]: @Deniallugo  @StanislavBreadless  most likely they are different.  If tx is finalized, more fields in the recept object might be filled.(at least the status of tx will be `finalized`)

File: docs/api/js/accounts-l1-l2.md (hunk 5 above)
Comment (AnastasiiaVashchuk) [reply docs/api/js/accounts-l1-l2.md:202]: @StanislavBreadless  I have a chat with @vladbochok  and he told me that the ` This value has no meaning for the `Deque` type of queue, but it will be used to prioritize the transactions that get into the `Heap` or `HeapBuffer` queues. ` is outdated now

Comment (StanislavBreadless) [reply docs/api/js/accounts-l1-l2.md:202]: @AnastasiiaVashchuk it is outdated because `Deque`/`Heap` are no longer present (we only have `Deque` rn). But the value is still unused, so please at least say that it is not needed right now.

File: docs/api/js/utils.md (hunk 12 above)
Comment (StanislavBreadless) [reply docs/api/js/utils.md:21]: I would rather put it like, "while Ether is actually a token deployed on the address `0x000000000000000000000000000000000000800a`, we also allow a more friendly zero address alias in our SDK and API"

File: docs/api/js/getting-started.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply docs/api/js/getting-started.md:126]: @AnastasiiaVashchuk agree

File: docs/api/js/types.md (hunk 2 above)
Comment (Deniallugo) [reply docs/api/js/types.md:6]: The whole snippet looks redundant. 
You copy and paste the file from js SDK.

File: docs/api/js/accounts-l1-l2.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply docs/api/js/accounts-l1-l2.md:35]: @Deniallugo @AnastasiiaVashchuk TBH, either option looks fine to me. Maybe it is better to keep `?` for consiceness

File: docs/api/js/accounts.md [hunk 16]
```diff
@@ -288,14 +285,14 @@ async withdraw(transaction: {
...
-| bridgeAddress (optional) | The address of the bridge contract to be used.                      |
-| overrides (optional)     | Transaction overrides, such as `nonce`, `gasLimit` etc. |
-| returns                  | A `TransactionResponse` object                                      |
+| Name                     | Description                                             |
+| ------------------------ | ------------------------------------------------------- |
+| tx.to                    | The address of the recipient on L1.                     |
+| tx.amount                | The amount of the token to transfer.                    |
+| token?         | The address of the token. `ETH` by default.             |
+| bridgeAddress? | The address of the bridge contract to be used.          |
+| overrides?     | **Ethereum** transaction overrides. May be used to pass `gasLimit`, `gasPrice`, etc.    |
```
[the remark this answers is not in the document]
Comment (Deniallugo) [reply docs/api/js/accounts.md:305]: Because withdraw is l2 transaction it has no l1 interaction for the user. 

The only withdrawal-related interaction l1 is done with the withdrawal finalizer.

File: docs/api/js/types.md (hunk 2 above)
Comment (AnastasiiaVashchuk) [reply docs/api/js/types.md:6]: it was before like this. It is not my feature...so I think we need to make changes for it in the separate PR

File: docs/api/js/accounts-l1-l2.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (Deniallugo) [reply docs/api/js/accounts-l1-l2.md:35]: conciseness is bad when we are talking about documentation. 
I have no strong opinion in this case. 
But I'd explicit documentation, rather than implicit

File: docs/api/js/types.md (hunk 2 above)
Comment (Deniallugo) [reply docs/api/js/types.md:6]: But it's still weird to have such snippets in the documentation. if we have an opportunity to have todos in our docs it definitely deserves a todo

Comment (AnastasiiaVashchuk) [reply docs/api/js/types.md:6]: agree.
will add now

File: docs/api/js/accounts-l1-l2.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (vladbochok) [reply docs/api/js/accounts-l1-l2.md:35]: Put my 5 cents, in this case conciseness  looks good, but in general I agree with @Deniallugo

File: docs/api/js/types.md [hunk 17]
```diff
@@ -1,7 +1,7 @@
 # Types
 
 All the types which are used in the SDK are referenced here:
-
+<!-- TODO: prepare quality documentation for SDK types with proper description and explanation -->
```
Comment (StanislavBreadless) [inline docs/api/js/types.md:4]: not sure if the doc comments inside the code are not enough (if there are no comments there, then we should add those)

If you still think better documentation is needed, then, please create a task in linear for it

File: docs/api/js/accounts.md [hunk 18]
```diff
@@ -131,8 +129,15 @@ const zkSyncProvider = new zksync.Provider("https://zksync2-testnet.zksync.dev")
 const ethereumProvider = ethers.getDefaultProvider("goerli");
 const wallet = new Wallet(PRIVATE_KEY, zkSyncProvider, ethereumProvider);
 
-const l1Contract = await zksync.Wallet.getL1BridgeContracts();
-console.log(l1Contract.address);
+const contract = await wallet.getMainContract();
+console.log(contract.address);
+```
+
+### Getting У
```
Comment (StanislavBreadless) [inline docs/api/js/accounts.md:136]: ?

File: docs/api/js/accounts.md [hunk 19]
```diff
@@ -111,7 +110,7 @@ const wallet = unconnectedWallet.connect(provider).connectToL1(ethProvider);
 ### Getting the zkSync L1 smart contract
 
 ```typescript
-async getMainContract(): Promise<Contract>
+async Adapter.getMainContract(): Promise<IZkSync>
```
Comment (Deniallugo) [inline docs/api/js/accounts.md:113]: Looks weird, what is Adapter?

File: docs/api/js/accounts.md [hunk 20]
```diff
@@ -111,29 +110,23 @@ const wallet = unconnectedWallet.connect(provider).connectToL1(ethProvider);
...
-
-const PRIVATE_KEY = "0xc8acb475bb76a4b8ee36ea4d0e516a755a17fad2e84427d5559b37b544d9ba5a";
-
-const zkSyncProvider = new zksync.Provider("https://zksync2-testnet.zksync.dev");
-const ethereumProvider = ethers.getDefaultProvider("goerli");
-const wallet = new Wallet(PRIVATE_KEY, zkSyncProvider, ethereumProvider);
-
-const contract = await wallet.getMainContract();
-console.log(contract.address);
+const l1Contract = await zksync.Wallet.getL1BridgeContracts();
```
[the remark this answers is not in the document]
Comment (AnastasiiaVashchuk) [reply docs/api/js/accounts.md]: it is outdated @Deniallugo

File: docs/api/js/accounts.md (hunk 19 above)
Comment (AnastasiiaVashchuk) [reply docs/api/js/accounts.md:113]: it is ts class..

Comment (Deniallugo) [reply docs/api/js/accounts.md:113]: We have neither import nor deffinition. 
From the user's perspective, it's just a random word, which they should find by search or where? 
How they can initialize this class? is it ethers class or zksync?

File: docs/api/js/getting-started.md [hunk 21]
```diff
@@ -90,7 +90,7 @@ const depositReceipt = await deposit.wait();
 // Retreiving the current (committed) zkSync ETH balance of an account
 const committedEthBalance = await zkSyncWallet.getBalance(zksync.utils.ETH_ADDRESS);
 
-// Retrieving the ETH balance of an account in the last finalized zkSync block.
+// Retrieving the ETH balance of an account in the last finalized block( #confirmations-and-finality)
```
Comment (StanislavBreadless) [inline docs/api/js/getting-started.md:93]: I don't think that the link will work

**Review by StanislavBreadless — APPROVED** [review]

**Review by krofax — APPROVED** [review]
